tests/e2e/Services/Databases/DatabasesBase.php

```diff
@@ -297,6 +297,80 @@ public function testCreateAttributes(array $data): array
         return $data;
     }
 
+    /**
+     * @depends testCreateDatabase
+     */
+    public function testNumericOnlyAttributes(array $data): void
+    {
+        $databaseId = $data['databaseId'];
+
+        $numericKeyCollection = $this->client->call(Client::METHOD_POST, '/databases/' . $databaseId . '/collections', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey']
+        ]), [
+            'collectionId' => ID::unique(),
+            'name' => 'Numeric Keys',
+            'documentSecurity' => true,
+            'permissions' => [
+                Permission::create(Role::user($this->getUser()['$id'])),
+            ],
+        ]);
+
+        $numericKeyId = $numericKeyCollection['body']['$id'];
+
+        $numOne = $this->client->call(Client::METHOD_POST, '/databases/' . $databaseId . '/collections/' . $numericKeyId . '/attributes/string', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey']
+        ]), [
+            'key' => '1',
+            'size' => 256,
+            'required' => false,
+            'array' => false,
+        ]);
+
+        // wait for database worker to create attributes
+        sleep(2);
+
+        // Create document with numeric key
+        $document = $this->client->call(Client::METHOD_POST, '/databases/' . $databaseId . '/collections/' . $numericKeyId . '/documents', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'documentId' => ID::unique(),
+            'data' => [
+                '1' => 'one',
+            ],
+            'permissions' => [
+                Permission::read(Role::user($this->getUser()['$id'])),
+                Permission::update(Role::user($this->getUser()['$id'])),
+                Permission::delete(Role::user($this->getUser()['$id'])),
+            ]
+        ]);
+
+        $id = $document['body']['$id'];
+
+        $this->assertEquals($document['body']['1'], 'one');
+
+        // Update document
+        $document = $this->client->call(Client::METHOD_PATCH, '/databases/' . $databaseId . '/collections/' . $numericKeyId . '/documents/' . $id, array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'data' => [
+                '1' => 'one updated',
+            ],
+            'permissions' => [
+                Permission::read(Role::users()),
+                Permission::update(Role::users()),
+                Permission::delete(Role::users()),
+            ],
+        ]);
+
+        $this->assertEquals($document['body']['1'], 'one updated');
+    }
+
```

Comment:
Could you add two more tests, please?
one for find and second for getDocument , while trying to use the select query operator for this specific attribute?
for checking validations are passed.

Reply:
@fogelito  this can be solved using + operator as well although I think we should first discuss if we want to allow attribute with numeric names or not

Reply:
I agree 👍